### ISSUE TITLE
Investigate clippy failure on example-shader with glam

### ISSUE THREAD

**Issue by repi** [EmbarkStudios/rust-gpu#186]
Running clippy on `example-shader` after the glam conversion in #149 results in:

```
     Compiling example-shader v0.1.0 (/home/runner/work/rust-gpu/rust-gpu/examples/example-shader)
  error: constant runtime array value
```

https://github.com/EmbarkStudios/rust-gpu/pull/149/checks?check_run_id=1326417262

it is from our codegen but not sure how we are getting it and needs further investigation

**Comment by khyperia** [comment]
The actual failure is due to panicking with a string, and strings are not supported yet. I added a quick ignored test as a reminder to implement it eventually. https://github.com/EmbarkStudios/rust-gpu/blob/0b5ddb697b842a1be6ffb01f6bbf49b7429c46b1/crates/spirv-builder/src/test/basic.rs#L37-L48

But for *why* clippy generates code that panics with a string, but normal builds don't, that I have no idea on. I tried investigating for a few hours, and came up with nothing.
